legacy/firmware/config.c

```diff
@@ -772,12 +784,69 @@ bool config_changeWipeCode(const char *pin, const char *wipe_code) {
   return sectrue == ret;
 }
 
-const uint8_t *session_getSessionId(void) {
-  if (!sessionIdCached) {
-    random_buffer(sessionId, 32);
+uint8_t session_findLRU(void) {
+  uint8_t lowest_index = MAX_SESSIONS_COUNT;
+  uint32_t lowest = sessionLRUCounter;
+  for (uint8_t i = 0; i < MAX_SESSIONS_COUNT; i++) {
+    if (sessionsCache[i].usage < lowest) {
+      lowest = sessionsCache[i].usage;
+      lowest_index = i;
+    }
+  }
+  // TODO: abort if lowest_index == MAX_SESSIONS_COUNT?
+  // TODO: it should never happen, so it is more of "internal" check
```

Comment:
Maybe not worth it?

Reply:
It will happen if `sessionLRUCounter` overflows.

Reply:
Indeed. Since `sessionLRUCounter` is 32 bytes, it seems like a fair assumption this should not happen?

Reply:
Oh, I am not worried about regular use, but attacker could use this. I don't know how many messages per second we can process on the USB, so I don't know how realistic that is.

Reply:
`ensure(lowest_index < MAX_SESSIONS_COUNT, NULL)`

Reply:
Great. Since `ensure` uses `secbool` I have used `ensure(sectrue * (oldest_index < MAX_SESSIONS_COUNT), NULL);`

Reply:
Ok, great. `ensure` as defined in `common.h` uses `secbool`, so I will use `ensure(sectrue * (oldest_index < MAX_SESSIONS_COUNT), NULL);` correct<

Reply:
Yes, you are right.